src/zenml/integrations/integration.py

```diff
@@ -62,53 +63,64 @@ def check_installation(cls) -> bool:
+                    extra_list = extras[1:-1].split(",")
+                    for extra in extra_list:
+                        try:
+                            requirements = dist.requires(extras=[extra])  # type: ignore[arg-type]
+                        except pkg_resources.UnknownExtra as e:
+                            logger.debug("Unknown extra: " + str(e))
+                            return False
+                        deps.extend(requirements)
                 else:
-                    logger.debug(
-                        f"Invalid requirement format '{r}' for integration {cls.NAME}."
-                    )
-                    return False
-
-            logger.debug(
-                f"Integration {cls.NAME} is installed correctly with "
-                f"requirements {cls.get_requirements()}."
-            )
-            return True
-        except pkg_resources.DistributionNotFound as e:
-            logger.debug(
-                f"Unable to find required package '{e.req}' for "
-                f"integration {cls.NAME}."
-            )
-            return False
-        except pkg_resources.VersionConflict as e:
-            logger.debug(
-                f"VersionConflict error when loading installation {cls.NAME}: "
-                f"{str(e)}"
-            )
-            return False
+                    deps = dist.requires()
+
+                for ri in deps:
+                    try:
+                        pkg_resources.get_distribution(ri)
+                    except pkg_resources.DistributionNotFound as e:
+                        logger.debug(
+                            f"Unable to find required dependency "
+                            f"'{e.req}' for requirement '{r}' "
+                            f"necessary for integration '{cls.NAME}'."
+                        )
+                        return False
+                    except pkg_resources.VersionConflict as e:
+                        logger.debug(
+                            f"Package version '{e.dist}' does not match "
+                            f"version '{e.req}' required by '{r}' "
+                            f"necessary for integration '{cls.NAME}'."
+                        )
+                        return False
+
+            except pkg_resources.DistributionNotFound as e:
+                logger.debug(
+                    f"Unable to find required package '{e.req}' for "
+                    f"integration {cls.NAME}."
+                )
+                return False
+            except pkg_resources.VersionConflict as e:
+                logger.debug(
+                    f"Package version '{e.dist}' does not match version "
+                    f"'{e.req}' necessary for integration {cls.NAME}."
+                )
+                return False
+
+        logger.debug(
+            f"Integration {cls.NAME} is installed correctly with "
+            f"requirements {cls.get_requirements()}."
+        )
+        return True
```

Comment:
Even a blind squirrel is right twice a day: _Dependency Caching: Repeated calls to pkg_resources.get_distribution can be optimized by caching the results of previous checks. This would reduce the overhead of querying installed packages multiple times for the same requirement._. Not a bad suggestion.